cli/src/ceremonies/commands/ceremonies_command_utils.rs

```diff
@@ -15,15 +15,375 @@
 
 */
 
+use crate::{
+	command_utils::get_chain_api, trusted_command_utils::get_pair_from_str,
+	trusted_commands::TrustedArgs, trusted_operation::perform_trusted_operation, Cli,
+};
 use codec::{Decode, Encode};
+use encointer_ceremonies_assignment::assignment_fn_inverse;
 use encointer_primitives::{
-	ceremonies::ProofOfAttendance, communities::CommunityIdentifier, scheduler::CeremonyIndexType,
+	ceremonies::{
+		Assignment, AssignmentCount, CommunityCeremony, MeetupIndexType, ParticipantIndexType,
+		ParticipantType, ProofOfAttendance,
+	},
+	communities::{CommunityIdentifier, Location},
+	scheduler::CeremonyIndexType,
+};
+use ita_stf::{
+	AccountId, KeyPair, Moment, PublicGetter, Signature, TrustedGetter, TrustedOperation,
 };
-use ita_stf::{AccountId, Signature};
+use itp_node_api::api_client::encointer::EncointerApi;
 use log::*;
+use serde::{Deserialize, Serialize};
 use sp_application_crypto::Ss58Codec;
 use sp_core::{sr25519 as sr25519_core, Pair};
 
+pub const ONE_DAY: Moment = 86_400_000;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Meetup {
+	pub index: MeetupIndexType,
+	pub location: Location,
+	pub time: Moment,
+	pub registrations: Vec<(AccountId, ParticipantType)>,
+}
+
+impl Meetup {
+	pub fn new(
+		index: MeetupIndexType,
+		location: Location,
+		time: Moment,
+		registrations: Vec<(AccountId, ParticipantType)>,
+	) -> Self {
+		Self { index, location, time, registrations }
+	}
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommunityCeremonyStats {
+	pub community_ceremony: CommunityCeremony,
+	pub assignment: Assignment,
+	pub assignment_count: AssignmentCount,
+	pub meetup_count: MeetupIndexType,
+	pub meetups: Vec<Meetup>,
+}
+
+impl CommunityCeremonyStats {
+	pub fn new(
+		community_ceremony: CommunityCeremony,
+		assignment: Assignment,
+		assignment_count: AssignmentCount,
+		meetup_count: MeetupIndexType,
+		meetups: Vec<Meetup>,
+	) -> Self {
+		Self { community_ceremony, assignment, assignment_count, meetup_count, meetups }
+	}
+}
+
+pub fn get_ceremony_stats(
+	cli: &Cli,
+	trusted_args: &TrustedArgs,
+	arg_who: &str,
+	community_identifier: CommunityIdentifier,
+	ceremony_index: CeremonyIndexType,
+) -> Option<CommunityCeremonyStats> {
+	let api = get_chain_api(cli);
+	let who = get_pair_from_str(trusted_args, arg_who);
+
+	let top: TrustedOperation = TrustedGetter::ceremonies_assignments(
+		who.public().into(),
+		community_identifier,
+		ceremony_index,
+	)
+	.sign(&KeyPair::Sr25519(who))
+	.into();
+	let encoded_assignments = perform_trusted_operation(cli, trusted_args, &top);
+	let assignment = decode_to_option(encoded_assignments).unwrap_or_default();
```

Comment:
I don't really understand the `decode_to_option`, why do we do that, shouldn't we just return an error?

Reply:
You're right. It is not necessary to "catch" the error at this step.